### PR TITLE
Added kernel panic files + Makefile fix for GCC 10

### PR DESCRIPTION
**Pull request by A52-bit**
I haven't implemented this function on the exception handler because:
1. The exception handler is too unstable, even turning off optimizations can crash the OS
2. My function is unstable, it crashed KVM several times

### CODE REVIEW

**Comment by AtieP** [comment]
That's not how you get registers, when invoking that it will only get them at the time you execute the function, not when the interrupt has been raised

**Comment by A52-bit** [comment]
@AtieP I know, this function is not called by an ISR, so I can't get the registers

File: src/misc/register.c [hunk 1]
```diff
@@ -0,0 +1,54 @@
+#include <register.h>
+#include <stddef.h>
+#include <halt.h>
+
+void getregisters(bool is_system_dead)
```
Comment (unsafecast) [inline src/misc/register.c:5]: I recommend using an enum for this kinda stuff, booleans are hard to read in a function call.

```
typedef enum {
    SYSTEM_STATE_DEAD,
    SYSTEM_STATE_ALIVE,
} system_state_t;
```

Comment (A52-bit) [reply src/misc/register.c:5]: ok, fixing it

Comment (AtieP) [reply src/misc/register.c]: why typedef it?

**Comment by AtieP** [comment]
> @AtieP I know, this function is not called by an ISR, so I can't get the registers

then it's pretty much pointless

**Comment by A52-bit** [comment]
@AtieP The code is totally portable to work with ISRs. It isn't implemented on the kernel because I haven't ported it yet.

File: src/misc/register.c (hunk 1 above)
Comment (A52-bit) [reply src/misc/register.c]: @AtieP Because GCC is weird and will return an error without the typedef

**Comment by unsafecast** [comment]
Also sorry for splitting the review in two :(

**Comment by A52-bit** [comment]
Looks very bad, I will close it

**Pull request closed without merging**